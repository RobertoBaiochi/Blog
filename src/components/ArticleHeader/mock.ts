import { ArticleHeaderProps } from '.';
import { data } from '../../api/data.json';

// const { posts } = data;
// const {
//   id,
//   attributes: { title, excerpt, createdAt },
// } = posts.data[1];

// export default {
//   id,
//   title,
//   excerpt,
//   cover: data.posts.data[1].attributes.cover,
//   author: data.posts.data[1].attributes.author.data,
//   categories: data.posts.data[1].attributes.categories.data,
//   createdAt,
// } as unknown as ArticleHeaderProps;

const {
  id,
  attributes: { title, excerpt, createdAt },
} = data.posts.data[1];

export default {
  id,
  title,
  excerpt,
  cover: data.posts.data[1].attributes.cover,
  author: data.posts.data[1].attributes.author.data,
  categories: data.posts.data[1].attributes.categories.data[1],
  createdAt,
} as unknown as ArticleHeaderProps;
