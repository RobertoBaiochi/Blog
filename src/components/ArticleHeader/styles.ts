import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacing.xlarge};
    margin-bottom: ${theme.spacing.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacing.small};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacing.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `};
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    margin-bottom: ${theme.spacing.medium};
  `};
`;
