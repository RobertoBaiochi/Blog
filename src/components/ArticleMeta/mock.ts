import { ArticleMetaProps } from './';
import { data } from '../../api/data.json';

const { posts } = data;

const {
  attributes: {
    createdAt,
    author: {
      data: {
        id: idAuthor,
        attributes: { displayName: displayNameAuthor, slug: slugAuthor },
      },
    },
    categories,
  },
} = posts.data[1];

export default {
  createdAt,
  author: {
    idAuthor,
    displayNameAuthor,
    slugAuthor,
  },
  categories,
} as unknown as ArticleMetaProps;

// import { ArticleMetaProps } from '.';

// export default {
//   createdAt: '2023-05-09T04:49:38.694Z',
//   author: {
//     id: '1',
//     displayName: 'Roberto Baiochi',
//     slug: 'roberto-baiochi',
//   },
//   categories: [
//     {
//       id: '1',
//       displayName: 'ReactJs',
//       slug: 'react-js',
//     },
//     {
//       id: '2',
//       displayName: 'NextJs',
//       slug: 'next-js',
//     },
//   ],
// } as ArticleMetaProps;
