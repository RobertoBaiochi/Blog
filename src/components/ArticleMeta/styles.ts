import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    width: 100%;
  `}
`;
