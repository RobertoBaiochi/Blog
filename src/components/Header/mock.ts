import { HeaderProps } from '.';
import { data } from '../../api/data.json';

const { setting } = data;
const {
  data: {
    id,
    attributes: { blogName, blogDescription },
  },
} = setting;

export default {
  id,
  blogName,
  blogDescription,
  logo: data.setting.data.attributes.logo.data,
} as unknown as HeaderProps;
