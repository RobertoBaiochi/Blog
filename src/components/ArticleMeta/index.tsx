import { formatDate } from '../../utils/format-date';
import { Author } from '../../shared-typed/author';
import { Category } from '../../shared-typed/category';
import { Wrapper } from './styles';

export type ArticleMetaProps = {
  createdAt: string;
  author?: Author;
  categories?: Category[];
};

const ArticleMeta = ({
  createdAt,
  author = undefined,
  categories = [],
}: ArticleMetaProps) => {
  return (
    <Wrapper>
      <p>
        {typeof author !== 'undefined' && (
          <>
            <span>Por </span>
            <a href={`/author/${author?.attributes.slug}`}>
              {author?.attributes.displayName}
            </a>
            <span className="separator"> | </span>
          </>
        )}

        <time dateTime={createdAt}>{formatDate(createdAt)}</time>

        {categories.length > 0 && (
          <>
            <span className="separator"> | </span>
            <span className="categories">
              {categories.map((category) => {
                const {
                  id,
                  attributes: { displayName, slug },
                } = category;

                return (
                  <span key={`article-meta-cat-${id}`}>
                    <a href={`/category/${slug}`}>{displayName}</a>
                  </span>
                );
              })}
            </span>
          </>
        )}
      </p>
    </Wrapper>
  );
};
export default ArticleMeta;
