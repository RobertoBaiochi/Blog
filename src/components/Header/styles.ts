import styled, { css } from 'styled-components';

import { Title as HeadingHeader } from '../Heading/styles';

export const HeaderContainer = styled.header`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.sizes.maxWidth};
    margin-inline: auto;
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacing.huge};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};

    ${HeadingHeader} {
      margin: 0 0 calc(${theme.spacing.small} - 3rem);
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    max-width: 48rem;
    height: 100%;
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacing.large};
  `};
`;
