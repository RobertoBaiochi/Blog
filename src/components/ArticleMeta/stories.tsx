import { Meta, StoryFn } from '@storybook/react';
import ArticleMeta, { ArticleMetaProps } from '.';

import mock from './mock';

export default {
  title: 'ArticleMeta',
  component: ArticleMeta,
  args: mock,
  argTypes: {
    createdAt: {
      control: {
        type: 'date',
      },
    },
  },
} as Meta<ArticleMetaProps>;

export const Template: StoryFn<ArticleMetaProps> = (args) => {
  return (
    <div>
      <ArticleMeta {...args} />
    </div>
  );
};

export const NoAuthor: StoryFn<ArticleMetaProps> = (args) => {
  return (
    <div>
      <ArticleMeta {...args} author={undefined} />
    </div>
  );
};

export const NoCategories: StoryFn<ArticleMetaProps> = (args) => {
  return (
    <div>
      <ArticleMeta {...args} categories={undefined} />
    </div>
  );
};
