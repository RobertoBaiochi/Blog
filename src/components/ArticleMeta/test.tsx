import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import ArticleMeta, { ArticleMetaProps } from '.';
import '@testing-library/jest-dom';

import mock from './mock';

const props: ArticleMetaProps = mock;

describe('<ArticleMeta />', () => {
  it('should render author and category links', () => {
    renderTheme(<ArticleMeta {...props} />);
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const { displayName, slug } = props.categories![0].attributes;
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const { displayName: authorName, slug: authorSlug } =
      props.author!.attributes;

    const linkAuthor = screen.getByRole('link', {
      name: authorName,
    });
    const linkCategoryNext = screen.getByRole('link', {
      name: displayName,
    });

    expect(linkAuthor).toHaveAttribute('href', `/author/${authorSlug}`);
    expect(linkCategoryNext).toHaveAttribute('href', `/category/${slug}`);
  });

  it('should render formated date', () => {
    renderTheme(<ArticleMeta {...props} />);
    const date = screen.getByText('9 de mai. de 2023');

    expect(date).toHaveAttribute('datetime', props.createdAt);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleMeta {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with no author and categories', () => {
    const { container } = renderTheme(
      <ArticleMeta {...props} author={undefined} categories={undefined} />,
    );

    expect(container).toMatchSnapshot();
  });
});
