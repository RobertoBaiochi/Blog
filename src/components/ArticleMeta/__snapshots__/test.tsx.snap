// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleMeta /> should match snapshot 1`] = `
<div>
  <div
    class="sc-7c55c9a4-0 bRvXpo"
  >
    <p>
      <span>
        Por 
      </span>
      <a
        href="/author/aira-cruz"
      >
        Aira Cruz
      </a>
      <span
        class="separator"
      >
         | 
      </span>
      <time
        datetime="2023-05-09T04:49:38.694Z"
      >
        9 de mai. de 2023
      </time>
      <span
        class="separator"
      >
         | 
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/react-js"
          >
            ReactJs
          </a>
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`<ArticleMeta /> should match snapshot with no author and categories 1`] = `
<div>
  <div
    class="sc-7c55c9a4-0 bRvXpo"
  >
    <p>
      <time
        datetime="2023-05-09T04:49:38.694Z"
      >
        9 de mai. de 2023
      </time>
    </p>
  </div>
</div>
`;
